invokeai/app/api/routers/models.py

```diff
@@ -54,226 +66,70 @@ async def list_models() -> ModelsList:
-        #             print(f">> Using configured VAE assigned to {models_to_merge[0]}")
-        #             merged_model_config.update(vae=vae)
-
-        #         self.generate.model_manager.import_diffuser_model(
-        #             dump_path, **merged_model_config
-        #         )
-        #         new_model_list = self.generate.model_manager.list_models()
-
-        #         socketio.emit(
-        #             "modelsMerged",
-        #             {
-        #                 "merged_models": models_to_merge,
-        #                 "merged_model_name": model_merge_info["merged_model_name"],
-        #                 "model_list": new_model_list,
-        #                 "update": True,
-        #             },
-        #         )
-        #         print(f">> Models Merged: {models_to_merge}")
-        #         print(f">> New Model Added: {model_merge_info['merged_model_name']}")
-        #     except Exception as e:
-        #         self.handle_exceptions(e)
\ No newline at end of file
+@models_router.post(
+    "/",
+    operation_id="update_model",
+    responses={
+        201: {
+        "model_response": "Model added", 
+        },
+        202: {
+        "description": "Model submission is processing. Check back later."
+        }, 
+    },
+)
+async def update_model(
+    model_request: CreateModelRequest
+) -> CreateModelResponse:
+    """ Add Model """
+    try:
+        model_request_info = model_request.info
+        print(f">> Checking for {model_request_info}...")
+        info_dict = model_request_info.dict()
+
+        ApiDependencies.invoker.services.model_manager.add_model(
+            model_name=model_request.name,
+            model_attributes=info_dict,
+            clobber=True,
+        )
+        model_response = CreateModelResponse(name=model_request.name, info=model_request.info, status="success")
+
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
+    
+    return model_response
+
+
+@models_router.delete(
+    "/{model_name}",
+    operation_id="del_model",
+    responses={
+        204: {
+        "description": "Model deleted"
+        }, 
+        404: {
+        "description": "Model not found"
+        }
+    },
+)
+async def delete_model(model_name: str) -> None:
+    """Delete Model"""
+    model_names = ApiDependencies.invoker.services.model_manager.model_names()
+    model_exists = model_name in model_names
+    
+    try:
+        # check if model exists
+        print(f">> Checking for model {model_name}...")
+
+        if not model_exists:
+            print(f">> Model not found")
+            raise HTTPException(status_code=404, detail=f"Model '{model_name}' not found")                
+        
+        # delete model
+        print(f">> Deleting Model: {model_name}")
+        ApiDependencies.invoker.services.model_manager.del_model(model_name, delete_files=True)
+        print(f">> Model Deleted: {model_name}")
```

Comment:
I think you need to return a 204 explicitly.

Reply:
Indeed.